### PR TITLE
Update Doc for Intel XPU Profiling

### PR DESCRIPTION
**Pull request by louie-tsai**
Updated below two pages for Intel XPU
https://pytorch.org/docs/stable/torch.compiler_profiling_torch_compile.html
https://pytorch.org/docs/stable/profiler.html

cc @svekars @brycebortree @sekyondaMeta @AlannaBurke @tstatler

### CODE REVIEW

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 1]
```diff
@@ -4,9 +4,9 @@ Profiling to understand torch.compile performance
 What to use torch.profiler for:
 -------------------------------
 
-torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
+torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU, CPU and XPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
```
Comment (dvrogozh) [inline docs/source/torch.compiler_profiling_torch_compile.rst:7]:
```suggestion
torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU, CPU or XPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
```

**Review by dvrogozh — APPROVED** [review]
looks good

**Comment by dvrogozh** [comment]
> label "topic: not user facing"

How public documentation might not be user facing? What was the reason to add this label?

**Comment by dvrogozh** [comment]
> # This PR needs a release notes: label

You need to add "release notes" label it seems.

**Comment by atalman** [comment]
@louie-tsai  please fix lint

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 2]
```diff
@@ -4,9 +4,9 @@ Profiling to understand torch.compile performance
 What to use torch.profiler for:
 -------------------------------
 
-torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
+torch.profiler is helpful for understanding the performance of your program at a kernel-level granularity - for example, it can show graph breaks and GPU, CPU or XPU utilization at the level of the program. The data provided by the profiler can often help users understand where to investigate further to understand model performance.
 
-To understand kernel-level performance, other toosl exist. NVIDIA's ncu tool can be used, or :ref:`inductor's profiling tools <torchinductor-gpu-profiling>`.
+To understand kernel-level performance, other tools exist. NVIDIA's ncu tool, Intel® VTune™ Profiler or :ref:`inductor's profiling tools <torchinductor-gpu-profiling>` can be used.
```
Comment (malfet) [inline docs/source/torch.compiler_profiling_torch_compile.rst:9]: Thank you for fixing the typo, but following edit is somewhat inconsistent, as in one case cli tool name is used (which imo should be in backticks, i.e. `ncu`, but in other marketing name is used. If one wants to expand this list, I suggest it should cover all hardware that PyTorch supports and include links to vendors of those tools, something like the following
```suggestion
To understand kernel-level performance, other tools exist, such as :ref:`Nvidia Nsight compute tool<https://developer.nvidia.com/nsight-compute>`, :ref:`AMD Omnitrace<https://rocm.docs.amd.com/projects/omnitrace/en/latest/>`,  Intel® VTune™ Profiler or :ref:`inductor's profiling tools <torchinductor-gpu-profiling>` can be used.
```

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 3]
```diff
@@ -23,9 +23,12 @@ Basics of using torch.profiler and viewing traces
 
     import torch
     from torchvision.models import resnet18
+    
+    # user can select device to use: 'cuda', 'xpu', etc.
+    device = 'cuda'
```
Comment (malfet) [inline docs/source/torch.compiler_profiling_torch_compile.rst:28]:
```suggestion
    device = 'cuda'      # or 'cpu', 'xpu', etc.
```

**Review by malfet — CHANGES_REQUESTED** [review]
I'm somewhat uncomfortable with replacing every mention of GPU with `GPU/XPU`.
Should we use some more inclusive (and less abbreviated term) say `AI Accelerator` or `Hardware accelerator`? 

Also, at least to me, more pictures do not seem to convey a better message but just confuse the user and generic outline and information present in GPU and XPU screenshots are the same

**Comment by louie-tsai** [comment]
> I'm somewhat uncomfortable with replacing every mention of GPU with `GPU/XPU`. Should we use some more inclusive (and less abbreviated term) say `AI Accelerator` or `Hardware accelerator`?
> 
> Also, at least to me, more pictures do not seem to convey a better message but just confuse the user and generic outline and information present in GPU and XPU screenshots are the same

appreciated for that helps, and committed all your suggestions accordingly.

**Comment by louie-tsai** [comment]
> > I'm somewhat uncomfortable with replacing every mention of GPU with `GPU/XPU`. Should we use some more inclusive (and less abbreviated term) say `AI Accelerator` or `Hardware accelerator`?
> > Also, at least to me, more pictures do not seem to convey a better message but just confuse the user and generic outline and information present in GPU and XPU screenshots are the same
> 
> appreciated for that helps, and committed all your suggestions accordingly.

@malfet 
any input?

**Comment by louie-tsai** [comment]
@malfet  @zou3519 @atalman @jingxu10 
please help to merge it if no concern for the change.

thanks!

**Comment by louie-tsai** [comment]
@malfet  possible to approve the PR if all look good to you?

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 4]
```diff
@@ -52,17 +56,24 @@ Here, we observe:
 * CompiledFunction and CompiledFunctionBackward events, which correspond to the dynamo-compiled regions.
 * CPU events at the top, and GPU events at the bottom.
 
-**Flows between CPU and GPU events**
+**Flows between CPU and accelerator events**
 
-Every kernel on the GPU occurs after being launched by code running on the CPU. The profiler can draw connections (i.e. “flows”) between the GPU and CPU events to show which CPU event launched a GPU kernel. This is particularly helpful because, with a few exceptions, GPU kernels are launched asynchronously.
+Every kernel on the GPU/XPU occurs after being launched by code running on the CPU. The profiler can draw connections (i.e. “flows”) between the GPU/XPU and CPU events to show which CPU event launched a GPU/XPU kernel. This is particularly helpful because, with a few exceptions, GPU/XPU kernels are launched asynchronously.
```
Comment (malfet) [inline docs/source/torch.compiler_profiling_torch_compile.rst:61]:
```suggestion
Every kernel on the accelerator occurs after being launched by code running on the CPU. The profiler can draw connections (i.e. “flows”) between the accelerator and CPU events to show which CPU event launched a accelerator kernel. This is particularly helpful because, with a few exceptions, accelerator kernels are launched asynchronously.
```

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 5]
```diff
@@ -52,17 +56,24 @@ Here, we observe:
...
 
-To view a flow connection, click on a GPU kernel and click “ac2g”:
+To view a flow connection, click on a GPU and click “ac2g”:
```
Comment (malfet) [inline docs/source/torch.compiler_profiling_torch_compile.rst:63]: Why this line needs to be modified?

File: docs/source/profiler.rst [hunk 6]
```diff
@@ -42,4 +42,4 @@ Intel Instrumentation and Tracing Technology APIs
 .. for tracking purposes
 .. py:module:: torch.profiler.itt
 .. py:module:: torch.profiler.profiler
-.. py:module:: torch.profiler.python_tracer
\ No newline at end of file
+.. py:module:: torch.profiler.python_tracer
```
Comment (malfet) [inline docs/source/profiler.rst:45]: What was changed on that line?

**Review by malfet — CHANGES_REQUESTED** [review]
I still see number of
> Here is a XPU trace and user could see similar CPU and XPU events for XPU Kernel launch:

sections which do not seem to bring a lot of value to the equation. Why developer needs to see XPU version, but not say ROCm version of the same graph? IMO because they are pretty similar and switching from one accelerator to another would not alter workflow semantically.

Please remove those extra pictures

File: docs/source/profiler.rst (hunk 6 above)
Comment (louie-tsai) [reply docs/source/profiler.rst]: new line stuff there. removed accordlingly.

File: docs/source/torch.compiler_profiling_torch_compile.rst [hunk 7]
```diff
@@ -52,17 +56,24 @@ Here, we observe:
...
 * CPU events at the top, and GPU events at the bottom.
 
-**Flows between CPU and GPU events**
+**Flows between CPU and accelerator events**
 
-Every kernel on the GPU occurs after being launched by code running on the CPU. The profiler can draw connections (i.e. “flows”) between the GPU and CPU events to show which CPU event launched a GPU kernel. This is particularly helpful because, with a few exceptions, GPU kernels are launched asynchronously.
+Every kernel on the accelerator occurs after being launched by code running on the CPU. The profiler can draw connections (i.e. “flows”) between the accelerator and CPU events to show which CPU event launched a accelerator kernel. This is particularly helpful because, with a few exceptions, accelerator kernels are launched asynchronously.
 
-To view a flow connection, click on a GPU kernel and click “ac2g”:
+To view a flow connection, click on a GPU and click “ac2g”:
```
Comment (louie-tsai) [inline docs/source/torch.compiler_profiling_torch_compile.rst:63]:
```suggestion
To view a flow connection, click on a GPU kernel and click “ac2g”:
```

File: docs/source/torch.compiler_profiling_torch_compile.rst (hunk 5 above)
Comment (louie-tsai) [reply docs/source/torch.compiler_profiling_torch_compile.rst]: indeed, added it back accordingly.

**Comment by louie-tsai** [comment]
@malfet  hope the latest changes address your feedback. Could you approve the PR if all look good to you?

**Comment by kit1980** [comment]
Removed 2.6 milestone after discussion with the release team.

**Review by malfet — APPROVED** [review]
Thank you for the update. By the way, int 2.6, there are finally an API that returns available accelerator which could be used here

**Comment by louie-tsai** [comment]
@malfet 
Sorry for late update. I rebase to the latest main branch, and no CI issue anymore.
please help to merge if it looks good to you.

**Comment by louie-tsai** [comment]
@no-stale

**Comment by louie-tsai** [comment]
@malfet also possible to remove stale label. thanks

**Review by malfet — CHANGES_REQUESTED** [review]
@louie-tsai docs build failures are real

**Review by malfet — APPROVED** [review]

**Comment by guangyey** [comment]
@louie-tsai Could you rebase this PR to the latest viable/strict branch to make CI happy

**Comment by louie-tsai** [comment]
> @louie-tsai Could you rebase this PR to the latest viable/strict branch to make CI happy

sure. just did. will check the status.

**Comment by louie-tsai** [comment]
@guangyey 
it looks good to me after rebase. please help us to merge it if all look good to you.

**Comment by guangyey** [comment]
@louie-tsai Thanks for your update. Will land once Lint CI passes.
